app/controllers/application_controller.rb

```diff
@@ -12,32 +12,42 @@ def status
   private
 
   def fetch_pages
-    @pages = Page.all # For side menu
+    # List pages for side menu
+    @pages = Page.where(language: I18n.locale)
+    @pages = Page.where(language: I18n.default_locale) if @pages.length.zero?
   end
 
   def set_locale
     I18n.locale = I18n.available_locales.include?(params[:locale]&.to_sym) ? params[:locale] : I18n.default_locale
-
     @locale = languages[I18n.locale]
-    @available_locales = []
-    I18n.available_locales.map do |lang|
-      next if lang == I18n.locale
-      @available_locales << {
-        language: lang,
-        translation: languages[lang]
-      }
-    end
+    @available_locales = other_available_locales
   end
 
   def default_url_options(options = {})
     options.merge(locale: I18n.locale)
   end
 
   def languages
-    @languages ||= ActiveSupport::HashWithIndifferentAccess.new(
-      en: 'English',
-      es: 'Español',
-      fr: 'Français'
-    )
+    @languages ||= ActiveSupport::HashWithIndifferentAccess.new.tap do |langs|
```

Comment:
@FranckKe I am wondering if this method can be totally static and put in an initializer?